packages/manager/modules/pci/src/projects/project/notebooks/Notebook.class.js

```diff
@@ -0,0 +1,303 @@
+    }
+
+    return labels;
+  }
+
+  addLabel(label) {
+    const labelIndex = this.getLabelIndex(label);
+    if (labelIndex >= 0) {
+      this.removeLabel(label);
+    }
+
+    this.labels.push(label);
+  }
+
+  simulateAddLabel(label) {
+    const labels = [...this.labels];
+    const labelIndex = this.getLabelIndex(label);
+    if (labelIndex >= 0) {
+      labels[labelIndex].title = label.title;
+    } else {
+      labels.push(label);
+    }
+
+    return labels;
+  }
+
+  convertLabels(labels = null) {
+    return (labels || this.labels).reduce(
+      (accumulator, label) => ({ ...accumulator, [label.id]: label.title }),
+      {},
+    );
+  }
+
+  setState(state) {
+    set(this, 'status.state', state);
+  }
+
+  get state() {
+    return this.status?.state;
+  }
+
+  get name() {
+    return this.spec?.name;
+  }
+
+  get region() {
+    return this.spec?.region;
+  }
+
+  get unsecureHttp() {
+    return this.spec?.unsecureHttp;
+  }
+
+  get duration() {
+    return this.status?.duration;
+  }
+
+  get durationString() {
+    const duration = moment.duration(this.duration, 'seconds');
+    let durationString = duration.years() ? `${duration.years()}y` : '';
+    durationString =
+      duration.months() || durationString
+        ? `${durationString} ${duration.months()}m`
+        : durationString;
+    durationString =
+      duration.days() || durationString
+        ? `${durationString} ${duration.days()}d`
+        : durationString;
+    durationString =
+      duration.hours() || durationString
+        ? `${durationString} ${duration.hours()}h`
+        : durationString;
+    durationString = `${durationString} ${duration.minutes()}mn`;
+    return durationString;
+  }
+
+  get accessUrl() {
+    return this.status?.url;
+  }
+
+  get editor() {
+    return this.spec?.env?.editor;
+  }
+
+  get framework() {
+    return this.spec?.env?.framework;
+  }
+
+  get frameworkVersion() {
+    return this.spec?.env?.frameworkVersion;
+  }
+
+  get environment() {
+    return `${this.spec.env.frameworkId} - ${this.spec.env.frameworkVersion}`;
+  }
+
+  get stateInfo() {
+    const state = this.status.state.toUpperCase();
+    return {
+      state,
+      stateGroup: {
+        error: NOTEBOOK_STATES.ERROR.includes(state),
+        warning: NOTEBOOK_STATES.WARNING.includes(state),
+        success: NOTEBOOK_STATES.SUCCESS.includes(state),
+      },
+    };
+  }
+
+  get labels() {
+    return this.spec.labels;
+  }
+
+  set labels(labels) {
+    this.spec.labels = labels;
+  }
+
+  get cpu() {
+    return this.spec?.resources?.cpu;
+  }
+
+  get gpu() {
+    return this.spec?.resources?.gpu || 'none';
+  }
+
+  get memory() {
+    return this.spec.resources.memory;
+  }
+
+  get formattedRunningDuration() {
+    return moment.duration(this.status.duration, 'seconds').humanize();
+  }
+
+  get formattedCreationDate() {
+    return moment(this.createdAt).format('LLL');
+  }
+
+  get automationInfoLink() {
```

Comment:
it's better to add ovhSubsidiary as function parameter and remove coreConfig attribute, i don't think this class should have coreConfig as a dependency

Reply:
Good, fixed.